### PR TITLE
1169: CI: allocate gcc-7 build to new agent pool

### LINKED ISSUE #1169: Upgrade to paid provisioned resources in Azure cloud for some of the builds that are compute/memory/storage intensive
**What Needs to be Done?**

Start with gcc-7, due to storage requirements being exceeded. Then, move Intel/NVCC to get those builds to pass more quickly.

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #1169

### CODE REVIEW

File: ci/azure/azure-gcc-7-ubuntu-mpich.yml [hunk 1]
```diff
@@ -56,9 +56,15 @@ stages:
...
-      vmImage: 'ubuntu-18.04'
+      name: 'vt-agent-pool'
     timeoutInMinutes: 180
     steps:
+      - task: Bash@3
+        displayName: Install cmake
+        inputs:
+          targetType: 'inline'
+          script: |
+            docker login --username lifflander1 --password 55e8e524-2219-4369-8f3c-71599d6695c5
```
Comment (cz4rs) [inline ci/azure/azure-gcc-7-ubuntu-mpich.yml:67]: cleartext password

**Review by cz4rs — APPROVED** [review]
looks good as long as the `gcc-7` passes, please squash the commits / rewrite history a bit

**Comment by cz4rs** [comment]
minor note: the gcc-7 build stage seems pretty slow :/

**Comment by lifflander** [comment]
> minor note: the gcc-7 build stage seems pretty slow :/

Yes, I'm planning on sqashing. I just upgraded that build to 4 cores. Now it runs in 30 mins.

File: ci/azure/azure-gcc-7-ubuntu-mpich.yml (hunk 1 above)
Comment (lifflander) [reply ci/azure/azure-gcc-7-ubuntu-mpich.yml]: Yes, I'm still not sure how to solve this problem.

Comment (PhilMiller) [reply ci/azure/azure-gcc-7-ubuntu-mpich.yml]: Every CI platform has some means of injecting secrets in an environment variable or such without exposing them in the scripts or output.

Comment (cz4rs) [reply ci/azure/azure-gcc-7-ubuntu-mpich.yml]: some random links:
https://stefanscherer.github.io/how-to-use-secrets-in-azure-pipelines/
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#secret-variables

**Comment by lifflander** [comment]
Once I have the password issue fixed, I will merge this. I have to rebuild the base linux image to fix this with the credentials encrypted in it.

File: ci/azure/azure-gcc-7-ubuntu-mpich.yml (hunk 1 above)
Comment (lifflander) [reply ci/azure/azure-gcc-7-ubuntu-mpich.yml]: Yeah, it's easy to do in Azure pipelines, but not easy with the virtual scale set. I did manage to figure it out the solution though.

Comment (lifflander) [reply ci/azure/azure-gcc-7-ubuntu-mpich.yml]: I've rewritten the history so it no longer exists

Comment (PhilMiller) [reply ci/azure/azure-gcc-7-ubuntu-mpich.yml]: And then changed that password, one presumes?

**Pull request merged**